launcher/signature-discovery/client/client_test.go

```diff
@@ -0,0 +1,88 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/defaults"
+	"github.com/containerd/containerd/namespaces"
+)
+
+func pullPublicImage(ctx context.Context, imageRef string, cdClient *containerd.Client) (containerd.Image, error) {
```

Comment:
is there a way we can avoid actually pulling down from gcr?

Reply:
Removed this function and replaced the actual image with an image descriptor in `Client` struct.